presto-hudi/src/main/java/com/facebook/presto/hudi/HudiFileSkippingManager.java

```diff
@@ -0,0 +1,298 @@
+import org.apache.hudi.common.table.view.FileSystemViewManager;
+import org.apache.hudi.common.table.view.FileSystemViewStorageConfig;
+import org.apache.hudi.common.table.view.SyncableFileSystemView;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.common.util.hash.ColumnIndexID;
+import org.apache.hudi.metadata.HoodieTableMetadata;
+import org.apache.hudi.metadata.HoodieTableMetadataUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.DateType.DATE;
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.RealType.REAL;
+import static com.facebook.presto.common.type.SmallintType.SMALLINT;
+import static com.facebook.presto.common.type.TinyintType.TINYINT;
+import static com.facebook.presto.common.type.Varchars.isVarcharType;
+import static com.facebook.presto.parquet.predicate.PredicateUtils.isStatisticsOverflow;
+import static java.lang.Float.floatToRawIntBits;
+
+/**
+ * MetaDataTable based filesManager.
+ * list Hudi Table contents based on the
+ *
+ * <ul>
+ *   <li>Table type (MOR, COW)</li>
+ *   <li>Query type (snapshot, read_optimized, incremental)</li>
+ *   <li>Query instant/range</li>
+ * </ul>
+ *
+ * Support dataSkipping.
+ */
+public class HudiFileSkippingManager
+{
+    private static final Logger log = Logger.get(HudiFileSkippingManager.class);
+
+    private final HoodieTableQueryType queryType;
+    private final Option<String> specifiedQueryInstant;
+    private final HoodieTableMetaClient metaClient;
+    private final HoodieTableMetadata metadataTable;
+
+    protected transient volatile Map<String, List<FileSlice>> allInputFileSlices;
+
+    private transient volatile SyncableFileSystemView fileSystemView;
+
+    public HudiFileSkippingManager(
+            List<String> partitions,
+            String spillableDir,
+            HoodieEngineContext engineContext,
+            HoodieTableMetaClient metaClient,
+            HoodieTableQueryType queryType,
+            Option<String> specifiedQueryInstant)
+    {
+        this.queryType = queryType;
+        this.specifiedQueryInstant = specifiedQueryInstant;
+        this.metaClient = metaClient;
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder().enable(true).build();
+        this.metadataTable = HoodieTableMetadata
+                .create(engineContext, metadataConfig, metaClient.getBasePathV2().toString(), spillableDir, true);
+        prepareAllInputFileSlices(partitions, engineContext, metadataConfig, spillableDir);
```

Comment:
we could let `prepareAllInputFilesSlices` return `allInputFileSlices` as its computation result, then `allInputFileSlices` could be a final member.